CrowServe/CrowServe.h

```diff
@@ -114,25 +121,47 @@ class Socket {
                         if (bytes_received < 1) {
                             // TODO: error handling
                             std::cout << "error";
-                            return;
+                            break;
                         }
                         message = std::span<const char>{buffer, (u32)bytes_received};
                     }
 
                     switch (ProtocolType(main_header.protocol)){
                         case ProtocolType::ProtocolCrownLink: {
-                            m_crownlink_protocol.handle(message_header.message_type, message, handler);
+                            m_crownlink_protocol.handle(CrownLink::MessageType(message_header.message_type), message, handler);
                         } break;
                         case ProtocolType::ProtocolP2P: {
-                            m_p2p_protocol.handle(message_header.message_type, message, handler);
+                            //m_p2p_protocol.handle(P2P::MessageType(message_header.message_type), message, handler);
                         } break;
                     }
                 }
             }
         }};
     }
 
-    bool receive();
+template<typename T>
```

Comment:
This should be auto formatted, but there should be space between template and bracket.

Maybe we should try google style from clangformat with ident set to 4, I remember not liking some details, but it's probably better than me nitpicking things that do not matter.